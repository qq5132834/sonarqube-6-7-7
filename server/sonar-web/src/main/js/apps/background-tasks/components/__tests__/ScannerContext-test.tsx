/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
jest.mock('../../../../api/ce', () => ({
  getTask: jest.fn()
}));

import * as React from 'react';
import { mount, shallow } from 'enzyme';
import ScannerContext from '../ScannerContext';
import { click, doAsync } from '../../../../helpers/testUtils';

const getTask = require('../../../../api/ce').getTask as jest.Mock<any>;

const task = {
  componentName: 'foo',
  status: 'PENDING',
  id: '123',
  submittedAt: '2017-01-01',
  type: 'REPORT'
};

it('renders', () => {
  const wrapper = shallow(<ScannerContext onClose={jest.fn()} task={task} />);
  wrapper.setState({ scannerContext: 'context' });
  expect(wrapper).toMatchSnapshot();
});

it('closes', () => {
  const onClose = jest.fn();
  const wrapper = shallow(<ScannerContext onClose={onClose} task={task} />);
  click(wrapper.find('.js-modal-close'));
  expect(onClose).toBeCalled();
});

it('fetches scanner context on mount', () => {
  getTask.mockImplementation(() => Promise.resolve({ scannerContext: 'context' }));
  const wrapper = mount(<ScannerContext onClose={jest.fn()} task={task} />);
  expect(wrapper.state()).toEqual({});
  expect(getTask).toBeCalledWith('123', ['scannerContext']);
  return doAsync().then(() => {
    expect(wrapper.state()).toEqual({ scannerContext: 'context' });
  });
});
