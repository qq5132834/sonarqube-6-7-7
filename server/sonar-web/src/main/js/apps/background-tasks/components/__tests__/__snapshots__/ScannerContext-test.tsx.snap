// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal modal-large"
  closeTimeoutMS={0}
  contentLabel="scanner context"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="modal-overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <div
    className="modal-head"
  >
    <h2>
      background_tasks.scanner_context
      : 
      foo
       [
      background_task.type.REPORT
      ]
    </h2>
  </div>
  <div
    className="modal-body modal-container"
  >
    <pre
      className="js-task-scanner-context"
    >
      context
    </pre>
  </div>
  <div
    className="modal-foot"
  >
    <a
      className="js-modal-close"
      href="#"
      onClick={[Function]}
    >
      close
    </a>
  </div>
</Modal>
`;
